src/Microsoft.AspNetCore.Proxy/ProxyServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,26 @@
+﻿
+using System;
+using Microsoft.AspNetCore.Proxy;
+
+namespace Microsoft.Extensions.DependencyInjection
+{
+    public static class ProxyServiceCollectionExtensions
+    {
+        public static IServiceCollection AddProxy(this IServiceCollection services)
+        {
+            return services.AddSingleton(new ProxyService(new SharedProxyOptions()));
```

Comment:
Scoping does not help here, that would be per request, not per back end. Also, creating a new HttpClient on each request would cause major perf problems and port exhastion.